pkg/nfd-master/nfd-master.go

```diff
@@ -605,19 +605,74 @@ func (m *nfdMaster) nfdAPIUpdateOneNode(nodeName string) error {
 	return nil
 }
 
-func (m *nfdMaster) refreshNodeFeatures(cli *kubernetes.Clientset, nodeName string, annotations, labels map[string]string, features *nfdv1alpha1.Features) error {
+// validateExtendedResources validates extended resources and returns a map
+// of valid extended resources.
+func (m *nfdMaster) validateExtendedResources(features *nfdv1alpha1.Features, resourceLabels map[string]string) map[string]string {
+	extendedResources := make(map[string]string)
+	for extendedResource, capacity := range resourceLabels {
+		// Add possibly missing default ns
+		extendedResource = addNs(extendedResource, nfdv1alpha1.ExtendedResourceNs)
+		// Check if given NS is allowed
+		ns, _ := splitNs(extendedResource)
+		if isNamespaceDenied(ns, m.deniedNs.wildcard, m.deniedNs.normal) {
+			if _, ok := m.args.ExtraLabelNs[ns]; !ok {
+				klog.Errorf("Namespace %q is not allowed. Ignoring Extended Resource  %q\n", ns, extendedResource)
+				continue
+			}
+		}
+
+		// Dynamic Value
+		if strings.HasPrefix(capacity, "@") {
+			// Remove the @ prefix
+			capacity = strings.TrimPrefix(capacity, "@")
+			// capacity is a string in the form of attribute.featureset.elements
+			split := strings.SplitN(capacity, ".", 3)
+			if value, ok := features.Attributes[split[0]+"."+split[1]]; ok {
```

Comment:
I think we want to improve the error messages here a bit. Let's call `split[0]+"."+split[1]` _feature_ and  `split[2]` _element_. We want an error message if _feature_ is not found (we could have another error message of _feature_ is found from Flags or Instances instead of Attributes). We also want an error message if _element_ is not found inside the _feature_. Third error is then the one that the value is not qualified as a quantity,